lib/active_record/connection_adapters/postgis_adapter/spatial_table_definition.rb

```
@@ -12,14 +12,15 @@ def initialize(types, name, temporary, options, as, adapter)
         # super: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb#L320
         def new_column_definition(name, type, options)
           if (info = MainAdapter.spatial_column_options(type.to_sym))
+            if (limit = options.delete(:limit))
+              options.merge!(limit) if limit.is_a?(::Hash)
```

It's not intentional. I agree it'd be better to raise an error.